### PR TITLE
[WIP] [DO NOT MERGE] RHDEVDOCS 5235 security configuration for Pipelines pods

### PR DESCRIPTION
**Pull request by mramendi**
Version(s):

4.12+
pipelines-docs-1.12 (automatic cp to this one is likely to fail)

Issue:

RHDEVDOCS 5235

Link to docs preview:

https://64941--docspreview.netlify.app/openshift-enterprise/latest/cicd/pipelines/configuring-security-context-for-pods

QE review:
- [ ] QE has approved this change.

Additional information:

This  PR is fully reviewed. It is to be merged when Pipelines 4.12 goes GA, also to be cherry-picked manually to pipelines-docs-1.12 (automatic won't work, I plan to do it manually). Requesting a merge review now, so that the merge can be done with no further wait at GA time.

### CODE REVIEW

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (vdemeester) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:9]: security context (missing an s)

**Review by vdemeester — APPROVED** [review]

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (mramendi) [reply cicd/pipelines/configuring-security-context-for-pods.adoc]: thanks, fixed

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (ppitonak) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:9]: Do we need to mention `1.3.x` - it's an ancient version.

**Review by ppitonak — APPROVED** [review]

File: modules/op-running-pipeline-run-and-task-run-with-custom-scc-and-service-account.adoc
Comment (cbippley) [inline modules/op-running-pipeline-run-and-task-run-with-custom-scc-and-service-account.adoc:8]:
```suggestion
When using the `pipelines-scc` security context constraint (SCC) associated with the default `pipelines` service account, the pipeline run and task run pods might face timeouts. This happens because in the default `pipelines-scc` SCC, the `fsGroup.type` parameter is set to `MustRunAs`.
```
IBM Style Guide suggests using "might" instead of "may" in cases like this. https://www.ibm.com/docs/en/ibm-style?topic=word-usage#m

**Review by cbippley — COMMENTED** [review]
One nit, otherwise LGTM! Please wait for @opayne1 to look at it before requesting a merge review. Thanks!

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (concaf) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:9]: > use a non-privileged security context by default
this might be a bit misleading for customers - on one hand this might mean that the security context being used is not the "privileged" security context, or it might mean that the security context being used is not privileged by nature and hence pretty restrictive, which is not the case as the `pipelines-scc` is pretty relaxed and privileged.

i suggest removing this line entirely to avoid confusion:
> pods that result from a pipeline run or task run use a non-privileged security context by default.

File: modules/op-configuring-default-maximum-scc.adoc
Comment (concaf) [inline modules/op-configuring-default-maximum-scc.adoc:35]:
```suggestion
<1> `spec.platforms.openshift.scc.default` specifies the default SCC that {pipelines-shortname} attaches to all TaskRun and PipelineRun pods.
```
i'm suggesting this as the current line could be too much information for the user

**Review by concaf — APPROVED** [review]

File: modules/op-configuring-default-maximum-scc.adoc
Comment (mramendi) [reply modules/op-configuring-default-maximum-scc.adoc:35]: Can't really agree as the SA is visible and configurable, but I'll add a clarification.

File: modules/op-configuring-scc-namespace.adoc
Comment (opayne1) [inline modules/op-configuring-scc-namespace.adoc:8]:
```suggestion
You can configure the security context constraint (SCC) for all pods that {pipelines-shortname} creates for pipeline runs and task runs that you create in a particular namespace. This SCC must not be restricted less than the maximum SCC that you configured using the `TektonConfig` CR, in the `spec.platforms.openshift.scc.maxAllowed` spec.
```

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (opayne1) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:9]:
```suggestion
The default service account for pods that {pipeline-shortnames} starts is `pipeline`. The security context constraint (SCC) associated with the `pipeline` service account is `pipelines-scc` as defined in the following YAML specification:
```
I think "The `pipelines-scc` SCC is similar to the `anyuid` SCC" was a little confusing. I think you can remove it completely.

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (opayne1) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:25]: I am almost wondering if this `Bildah` information should be in a procedure module with more information on how you can make "the following changes" it seems out of place in this assembly, but I am not sure.

**Review by opayne1 — COMMENTED** [review]
A couple comments from me! Also, dont forget to squash your commits before sending to merge review.

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (mramendi) [reply cicd/pipelines/configuring-security-context-for-pods.adoc:25]: This information was there since the previous version, so until a deep restructure (which I hope to complete eventually) I would not want to move it.

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (mramendi) [reply cicd/pipelines/configuring-security-context-for-pods.adoc:9]: Unfortunately this is not true. The YAML specification specifices only the difference from the `anyuid` SCC. I have rephrased for clarity. (This text itself is there since the previous version)

File: modules/op-configuring-scc-namespace.adoc
Comment (mramendi) [reply modules/op-configuring-scc-namespace.adoc:8]: This one is very fair!

File: modules/op-configuring-default-maximum-scc.adoc
Comment (mramendi) [reply modules/op-configuring-default-maximum-scc.adoc:35]: I had to keep the name of the SA because we have docs about changing it elsewhere, I worked out some way.

**Comment by opayne1** [comment]
Thanks for the updates! Dont forget to squash all your commits to one commit before moving on to merge review.

**Comment by mramendi** [comment]
@opayne1 could you mark approval please?

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (ousleyp) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:9]: `{pipeline-shortnames}` should be `{pipelines-shortname}`

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (ousleyp) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:16]: Because this is a YAML snippet, I think the ellipses should be commented out. This is from the [openshift-docs/contributing_to_docs/doc_guidelines.adoc](https://github.com/openshift/openshift-docs/blob/main/contributing_to_docs/doc_guidelines.adoc) file:

```
For snippets or sections of a file, use an ellipsis (…​) to show that the file continues before or after the quoted block. For YAML, use ('#…​') instead.

apiVersion: v1
kind: Pod
metadata:
  labels:
    test: liveness
# ...
```

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (ousleyp) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:41]: Additional resources sections that apply only to specific modules within an assembly should use `.Additional resources` formatting, not `===`. They also do not need IDs, but they do need the `role` statement. Here's the relevant guideline from the contribution guide:

```
Only use . formatting (.Additional resources) in a module or to follow a module in an assembly. 
Because you cannot use the xrefs in modules, this functions as a trailing include at the assembly level, 
where the . formatting of the include statement indicates that the resource applies specifically to the 
module and not to the assembly. For example:

[role="_additional-resources"]
.Additional resources
* link:some-url.com[Human readable label]
* xref:some_xref[Human readable label]
* xref:some_other_xref[Human readable label]
```

File: cicd/pipelines/configuring-security-context-for-pods.adoc
Comment (ousleyp) [inline cicd/pipelines/configuring-security-context-for-pods.adoc:38]: Please add a line break after this include statement.

`// Add a blank line after each 'include::' statement.` Source: the [modular docs assembly template](https://raw.githubusercontent.com/redhat-documentation/modular-docs/master/modular-docs-manual/files/TEMPLATE_ASSEMBLY_a-collection-of-modules.adoc)

File: modules/op-configuring-scc-namespace.adoc
Comment (ousleyp) [inline modules/op-configuring-scc-namespace.adoc:13]: I'd add a `+` on this line so that it's clearer that the example is directly related to the step.
If possible, I would also suggest adding info about how to open this manifest for editing.

**Review by ousleyp — COMMENTED** [review]
It looks like you have a merge conflict to resolve, and I left a few minor comments. Otherwise, this is looking good. :) Thanks!

File: modules/op-configuring-scc-namespace.adoc
Comment (mramendi) [reply modules/op-configuring-scc-namespace.adoc:13]: Unfortunately can't cycle back to engineering at this stage; annotations are already referenced several times in the doc set, so this is realy an existing issue and I created a backlog Jira for it at https://issues.redhat.com/browse/RHDEVDOCS-5634

**Pull request closed without merging**
